### ISSUE TITLE
Codespace doesn't stop on exit

### ISSUE THREAD

**Issue by sdrzazga** [github/codespaces-jetbrains-feedback#77]
Hi.

When I exit a Codespace and chose the option `Close and Stop` ... the corresponding codespace keeps running/active.

`IDEA 2022.3 & GitHub Codespaces 0.3.0.1042`

**Comment by thomas-sickert** [comment]
Hey @sdrzazga, thanks for the report. That `Close and Stop` option is actually for the JetBrains remote dev stack (so it closes the Client and stops the Backend IDE). It is ambiguous, so I understand the confusion, sorry!

Improved flows around the codespace lifecycle will be added in the future but, in the meantime, all management tasks like stopping the codespace will need to be performed from https://github.com/codespaces or one of the our other clients.

**Comment by sdrzazga** [comment]
Hey @thomas-sickert .

Thanks for the quick response. That sure is confusing, since I was under the impression that it should stop the corresponding workspace as well.
I'll keep that in mind, but using JetBrains and Codespaces so far has been a very painful experience.

**Comment by thomas-sickert** [comment]
Sorry to hear that you're having a painful experience. Would you mind sharing some of the other pain points? It will help us prioritize future work and we can pass along feedback to JetBrains as well.

**Comment by sdrzazga** [comment]
Certainly.

Firstly, I have a hard time understanding when running into an issue what the right channel is to report it or check the corresponding documentation. In my case the is IDEA, JB Gateway and the GH Codespace plugin/provider.
It's confusing for me to tell them in practice apart.

2. JB Gateway ... I understand that for remote development you need certain providers aka plugins. Why are in `Settings -> Plugins -> Marketplace` plugins for Gateway such as `String Manipulation, `AWS Toolkit`, `IdeaVim` etc. ? It makes me wonder, if I really understand the purpose of this application.

3. Status of Codespaces which is shown in JG Gateway seems mostly out of sync at the beginning. GH say it active but Gateway suggests it's still starting up.
<img width="172" alt="Screenshot 2023-01-25 at 15 15 17" src="https://user-images.githubusercontent.com/17554464/214680296-8c847d3c-813e-486a-bbc4-07ab418ebdbf.png">

4. I did the "foolish" mistake recently of switching to the New UI. I love it and yes it is _BETA_ which so far has proven for me not really being an issue except for remote development using JB. The time it takes to start up a session is way too long and then once it's up I run into issues like [this](https://github.com/github/codespaces-jetbrains-feedback/issues/72) which required a restart of the IDE. Cannot tell how this experience here differs to the classic UI, but I won't go back.

5. Once I have a running session, I cannot get my usual settings/plugins synced since it is currently not [implemented](https://github.com/github/codespaces-jetbrains-feedback/issues/73). Manual "setup" of this IDE won't work for me. I need a method to sync or automate.

6. Settings in the remote IDE. I don't know, if it's just me but this is MEGA confusing:
<img width="221" alt="Screenshot 2023-01-25 at 07 42 11" src="https://user-images.githubusercontent.com/17554464/214682430-116f4dbe-08a4-456a-81d9-cb6e06264566.png">
It took me some time to find the corresponding documentation and even then I don't understand why there is a difference between `On Host` and something else.

There are just a few points which come to mind. If this was my first experience with JetBrains I would have perhaps never ever started using IDEA and become a paying customer.
I understand it is in _BETA_ state, but it feels more like an early _ALPHA_ . 
At this point, I have to familiarize myself with VSCode since this one works perfectly fine and is rocket-fast.

**Comment by sdrzazga** [comment]
I have to add a few things to my previous comments now that I spent some more time on this topic.

1. *Support for _dev containers_ *, I know this is not within the scope of this plugin. However, using VSCode I get nice control right from within my IDE to add more [features](https://containers.dev/features) to my dev container.
<img width="603" alt="Screenshot 2023-02-01 at 08 26 44" src="https://user-images.githubusercontent.com/17554464/216056159-3e503a1a-9190-439e-8004-0865552ca517.png">

2. *Interaction with the Codespace itself*, sorry to compare but in VSCode I can do that:
 
<img width="608" alt="Screenshot 2023-02-01 at 08 25 41" src="https://user-images.githubusercontent.com/17554464/216057317-feb1dbae-3a36-4ac4-bb03-42064f1a02c8.png">

I'm particularly looking for essentials like `View Creation Log`, `Rebuild Container` & `Full Rebuild Container`.
I haven't discovered that using Jetbrains. This should be added ASAP.
This would improve a GH Codespace experience. In particular, being able to have a look at the creation log is very useful when troubleshooting the environment.

**Comment by cmuto09** [comment]
Thank you for the feedback @sdrzazga!

> Firstly, I have a hard time understanding when running into an issue what the right channel is to report it or check the corresponding documentation. In my case the is IDEA, JB Gateway and the GH Codespace plugin/provider.
It's confusing for me to tell them in practice apart.

Totally agree. We work closely with the JetBrains team and share feedback from this repository with them when there are issues that land in their court. So in the immediate term feel free to put all issues here and we can communicate them to JetBrains.

In the future, how would you expect to be able to give feedback?

> Status of Codespaces which is shown in JG Gateway seems mostly out of sync at the beginning. GH say it active but Gateway suggests it's still starting up.

Great feedback- we will get this updated such that it polls for the latest state rather than fetching it only once

> I did the "foolish" mistake recently of switching to the New UI. I love it and yes it is BETA which so far has proven for me not really being an issue except for remote development using JB. The time it takes to start up a session is way too long and then once it's up I run into issues like https://github.com/github/codespaces-jetbrains-feedback/issues/72 which required a restart of the IDE. Cannot tell how this experience here differs to the classic UI, but I won't go back.

> Once I have a running session, I cannot get my usual settings/plugins synced since it is currently not https://github.com/github/codespaces-jetbrains-feedback/issues/73. Manual "setup" of this IDE won't work for me. I need a method to sync or automate.

> Settings in the remote IDE. I don't know, if it's just me but this is MEGA confusing

These are JetBrains related issues. I will share this feedback with them this week!

> *Support for dev containers *, I know this is not within the scope of this plugin. However, using VSCode I get nice control right from within my IDE to add more [features](https://containers.dev/features) to my dev container.

This is actually within the scope of this plugin, and we intend to add this functionality in the coming quarter or two. What would you consider higher value: codespace management (CRUD methods via the Gateway) or a dev container update wizard like this?

> I'm particularly looking for essentials like View Creation Log, Rebuild Container & Full Rebuild Container.
I haven't discovered that using Jetbrains. This should be added ASAP.

Very much agreed. You can view the creation log today in JetBrains [by using Codespaces plugin in the remote IDE](https://docs.github.com/en/codespaces/troubleshooting/github-codespaces-logs?tool=jetbrains). This is also available via the [Actions Menu in JetBrains](https://www.jetbrains.com/help/rider/Navigating_to_Action.html). If you open the actions menu and type `Codespaces:` you will see all the available actions.

Rebuild is currently not supported due to a blocker on the JetBrains side that they are actively working on. We understand the importance of this workflow and it is top priority for us- we will get it added as soon as JetBrains completes the blocking work on their side. I anticipate it will land this quarter.

Thank you so much for all the feedback @sdrzazga!

**Comment by cmuto09** [comment]
FYI: here is the [JetBrains YouTrack issue](https://youtrack.jetbrains.com/issue/GTW-2510/Merge-remote-and-local-plugins-to-a-single-page) regarding the plugin installation confusion (on host vs. on client) if you would like to track it or communicate with the JetBrains team on it

**Comment by sdrzazga** [comment]
Hello @cmuto09 .

I appreciate your response!

> In the future, how would you expect to be able to give feedback?

This is hard to answer. If a user would have enough transparency on what piece belongs to whom, similar to the JB IDE's & plugins. I'm ok to report when it comes to remote development `JB Gateway/IDE issues -> JB using Youtrack`, `Plugins -> Author (usually GH)` and `GH related issues -> GH`.
Just in the current situation, I have major difficulties to tell things apart.

>This is actually within the scope of this plugin, and we intend to add this functionality in the coming quarter or two. What would you consider higher value: codespace management (CRUD methods via the Gateway) or a dev container update wizard like this?

I'm a little bit confused here. [dev container](https://containers.dev) can be used locally or in other contexts such as GH Codespaces. In my head, JB would have to develop their own IDE plugin similar to MS to deal with this.
Sure, when it comes then to GH Codespaces there will be an overlap when everything has to come together, but I don't understand why this should be tackled in this plugin. Clearly, when I use dev container locally I shouldn't have to use the GH Codespaces plugin for JB.
Also in terms of priorities. Sure, that whole topic of CRUD around GH Codespaces at least when it comes to destroying a codespace from my PoV even with VSCode/GH Codespace "UI" is painful. Too many clicks when using UIs. I changed for this aspect completely to [GH CLI](https://cli.github.com).
Creation is straightforward and easy...to stop a codespace deliberately seems currently only to be an issue with JB. 
Ideally, I want to have down the road full control of this from within my IDE instead of switching back and forth.

So to sum this up. JB has to work on transparency for users to be able to tell responsibilities apart. A more "lightweight" backend which I believe is currently about 1GB. And the support of dev container.
